src/compiler/utilities.ts

```diff
@@ -2171,6 +2172,7 @@ export function createFileDiagnosticFromMessageChain(file: SourceFile, start: nu
         category: messageChain.category,
         messageText: messageChain.next ? messageChain : messageChain.messageText,
         relatedInformation,
+        canonicalHead: messageChain.canonicalHead,

```

Comment:
Does this need to be serialized in buildInfo when serializing diagnostic. Will we need this to deduplicate across multiple files?

Reply:
> Will we need this to deduplicate across multiple files?

I don't think so, this should only be relevant for deduplicating diagnostics of a single file.

> Does this need to be serialized in buildInfo when serializing diagnostic?
What's the criterion for needing to serialize this in `buildInfo`?

Reply:
you would need this info to deduplicate with errors from other files or errors emitted for same file in other phase (which obviously isn't true - like say same error is reported by declaration emit and semantic diagnostic just giving hypothetical example here)

Reply:
Given this doesnt get used beyond file boundaries this should be ok to be not serialized. 
That is checking files in any order will result in same error in that file (which we were anyways assuming till now) this is good.